### DISCUSSION TITLE
Şifre

### DISCUSSION THREAD

**Discussion by OmerAkbayrak** [sonsuzus/sonsuzus#392] [Soru Cevap]
### ****Problem****
Danny, Manny' nin facebook hesabının olası şifrelerine dair bir listeye sahip. Tüm şifrelerin uzunlukları farklı, ama Danny Manny'nin palindrom sayıların büyük bir hayranı olduğunu biliyor. Yani şifresi ve şifresinin tersten yazılışı da listede olmalı.

### **Girdi:**
Girdinin ilk satırı N sayısını yani listedeki olası şifre sayısını içerecek.
N tane satırdan her birinde tek bir kelime olacak. kelimelerin uzunluğu 2'den fazla 14'den az olacak, tüm karakterler küçük harfli olacak ve ingiliz alfabesinden olacak.

### **Çıktı:**
İlk ve tek satır doğru şifrenin harf sayısı ve şifrenin ortasındaki harfi içerecektir.

### **Sınırlamalar:**
1<N<100

### **Örnek Girdi:**                      
4                                              
abc
def
feg
cba
### **Örnek Çıktı:**
3 b 

_**Sorunun Linki: https://www.hackerearth.com/practice/algorithms/string-algorithm/basics-of-string-manipulation/practice-problems/algorithm/password-1/**_

> Contributer is:Lalit Kundu

**Comment by nilbuyukkinaci** [comment]
#include <iostream>
#include <string>
using namespace std;

string a[105], b[105], cevap;

int main() {
    int n;

    cin >> n;

    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    for (int i = 0; i < n; i++) {
        for (int j = a[i].length() - 1; j >= 0; j--) {
            b[i] += a[i][j];
        }
    }

    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (a[i] == b[j]) {
                cevap = a[i];
                break;
            }
        }
    }

    cout << cevap.length() << " " << cevap[cevap.length() / 2];
}
